core/src/main/java/org/apache/carbondata/core/datastore/block/SegmentProperties.java

```diff
@@ -147,6 +186,28 @@ private void fillBlockToDimensionOrdinalMapping() {
     }
   }
 
+  /**
+   * compare the segmentproperties based on fingerprinter
+   */
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof SegmentProperties)) {
+      return false;
+    }
+    SegmentProperties another = (SegmentProperties) obj;
```

Comment:
Please change the variable name to `segmentProperties`

Reply:
resolved